R/plotNorm.R

```diff
@@ -76,25 +80,141 @@ plotNorm <- function(DGEdata,
+}
+
+build_cx_density_data <- function(data) {
+    cx.data <- data.frame()
+    for (sample in levels(as.factor(data$SampleID))) {
+        if (nrow(cx.data) == 0) {
+            cx.data <- select_sample(data, sample, TRUE)
+        } else {
+            cx.data <-
+                cbind(cx.data, select_sample(data, sample))
+        }
+
     }
+    return(cx.data)
+}
+
+select_sample <- function(data, sampleID, select_all = FALSE) {
+    if (select_all) {
+        sample <-
+            subset(data, SampleID == sampleID,
+                   select = c("GeneID", "Normalization", "Log2CPM"))
+        colnames(sample)[3] = sampleID
+    } else {
+        sample <-
+            subset(data, SampleID == sampleID, select = c("Log2CPM"))
+        colnames(sample)[1] = sampleID
+    }
+
+    return(sample)
+}
+
+build_cx_density_plot <- function(data, title) {
+    cx.data <- build_cx_density_data(data)
+    xlab <- "Log2CPM"
+    ylab <- "density"
+    events <- htmlwidgets::JS(
+        "{ 'mousemove' : function(o, e, t) {
+                           if (o != null && o != false) {
+                             if (o.objectType == 'Density' &&
+                                o.display != null &&
+                                o.display.indexOf('-') > -1) {
+                                sampleID = o.display.split('-')[2];
+                                t.showInfoSpan(e,
+                                '<b>Sample ID</b>: ' + sampleID );
+                             } else {
+                               t.showInfoSpan(e, o.display);
+                             };
+                          }; }}")
+    resultPlot <- canvasXpress::canvasXpress(data                    = cx.data[,-c(1, 2)],
+                                             varAnnot                = cx.data[, c(1, 2)],
+                                             histogramData           = TRUE,
+                                             graphType               = "Scatter2D",
+                                             xAxisTitle              = xlab,
+                                             yAxisTitle              = ylab,
+                                             hideHistogram           = TRUE,
+                                             showHistogramDensity    = TRUE,
+                                             segregateVariablesBy    = list("Normalization"),
+                                             title                   = title,
+                                             showLegend              = FALSE,
+                                             colorScheme             = "GGPlot",
+                                             events                  = events)
+    return(resultPlot)
+}
+
+build_cx_box_plot <- function(data, title) {
+    cx.data <- build_cx_density_data(data)
+    xlab <- "Log2CPM"
+    ylab <- "SampleID"
+    y <- as.data.frame(t(cx.data[,-c(1, 2)]))
+    # events <- htmlwidgets::JS("{ 'mousemove' : function(o, e, t) {
```

Comment:
Don't keep any commented code in main functionality if it's not required remove it.